azkaban-common/src/main/java/azkaban/executor/KeyStoreManager.java

```diff
@@ -0,0 +1,40 @@
+package azkaban.executor;
+
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.security.KeyStore;
+
+/**
+ * KeyStoreManger is responsible to manage the KeyStore used during containerized execution of a
+ * flow. The HadoopSecurityManager must provide the KeyStore to CredentialProvider
+ */
+public class KeyStoreManager {
+  private static volatile KeyStoreManager ksmInstance = null;
+  private final static Logger logger = LoggerFactory.getLogger(KeyStoreManager.class);
+
+  private KeyStore keyStore = null;
+
+  public static KeyStoreManager getInstance() {
```

Comment:
Null check also needs to be within the synchronized block, after which effectively all the content will be in the synchronized block. Hence, better to keep the method synchronized and remove the synchronized block altogether. 
public static synchronized KeyStoreManager getInstance() ...